src/blockdata/script.rs

```diff
@@ -154,6 +154,22 @@ impl fmt::Display for Error {
 #[cfg_attr(docsrs, doc(cfg(feature = "std")))]
 impl ::std::error::Error for Error {}
 
+// Our internal error proves that we only return these two cases from `read_uint_iter`.
+// Since it's private we don't bother with trait impls besides From.
+enum UintError {
```

Comment:
Nit: I do not see any harm of doing derives here with just a single line. But pls do not discard other reviews just because of this one Nit

Reply:
The type is private (at least for now) and none of the derives is needed, so let's not needlessly blow up compile time. If we decide to make it public (I'm in favor of this) there will be all applicable derives and trait impls.

Reply:
"data types should eagerly implement derives" (from rust guidelines). :)

I am not insisting at all, just trying to understand the logic. Ok, the logic is not to blow the computing time.

However from my experience frequently these private error types become public and ppl forget to add derives to them - at least it cost me a lot of time to update `rust-secp256k1`, `rust-bitcoin` and minor crates with derives required to do a proper error handling on my own error types :(

Reply:
@dr-orlovsky I should have realized there's "public" word missing in that line when reviewing. Yeah, forgetting is unfortunate. Maybe have some checklists for various common tasks that could include this when changing a type to public? I'm thinking of something similar LND has.